### ISSUE TITLE
A better way to swap keys with keys (or improving langmap).

### ISSUE THREAD

**Issue by trusktr** [neovim/neovim#2199]
I've tried using `langmap` to do the following:

```viml
                set langmap=hjki;ihjk

                map <c-h> <c-left>
                map <c-j> <c-down>
                map <c-k> <c-up>
                map <c-l> <c-right>

                noremap <c-left> b
                noremap <c-down> 10<down>
                noremap <c-up> 10<up>
                noremap <c-right> e
```

but the langmap doesn't seem to apply to the `<c-*>` mappings. I was hoping to be able to press `ctrl`+`i` to be able to move up 10 lines hence the mappings `map <c-k> <c-up>` and `noremap <c-up> 10<up>`, but what ends up happening is that I still have to press `ctrl`+`k` to move up 10 lines.

I know it's possible to do

```viml
                noremap h i
                noremap j h
                noremap k j
                noremap i  k
```

but this means I have to change

```viml
                map <c-h> <c-left>
                map <c-j> <c-down>
                map <c-k> <c-up>
                map <c-l> <c-right>
```

to

```viml
                map <c-j> <c-left>
                map <c-k> <c-down>
                map <c-i> <c-up>
                map <c-l> <c-right>
```

In reality, I have a ton more mappings to get this `HJKL` to `IJKL` movement conversion to work in all the possible scenarios (f.e. `<c-w>i` to move up a window instead of `<c-w>k`, etc).

Ideally, I'd like for all my mappings to use the standard `HJKL`, then to be able to easily swap the physical meaning of the keys themselves just like `langmap` but everywhere (except INSERT and COMMAND).

I found out about langmap [here](http://stackoverflow.com/questions/7233505/how-to-swap-keys-in-vim-normal-mode-not-just-remap-them), and it seemed like it was going to be the solution, except for it's not working in all cases as with `<c-*>`.

It'd be nice to either fix langmap so it works in all cases as would be intuitively expected with `<c-*>`, or perhaps make these: changekey, nchangekey, xchangekey, ichangekey, etc. Basically, they'd match with the pattern seen with map, nmap, xmap, imap, etc, and might give ultimate control of changing keys before making any maps. chkey could be short for changekey, etc.

Then for example:

```viml
                changekey h i
                changekey j h
                changekey k j
                changekey i k

                map <c-h> <c-left>
                map <c-j> <c-down>
                map <c-k> <c-up>
                map <c-l> <c-right>

                noremap <c-left> b
                noremap <c-down> 10<down>
                noremap <c-up> 10<up>
                noremap <c-right> e
```

**Comment by trusktr** [comment]
I think just fixing langmap would be the simpler solution? A bonus could be to add nlangmap, vlangmap, xlangmap, etc.

**Comment by trusktr** [comment]
As a further example, although I've tried `map i k` which should cover "operator pending" mode, I still have to `map <c-w>i <c-w>k`. I would think that after having pressed `<c-w>` I'd be in Operator Pending mode, so I think that should be fixed also. If not, then there should at least be some "Other Mode" for that, which might also apply to `z*` and `g*` sequences. It doesn't make intuitive sense to me why z and g don't go into operator pending mode even though timeouts and such still apply, and they seem the same to operator pending in every way (internally they might not be, but that's not inherent to the user).

**Comment by trusktr-goin** [comment]
Seems to be fixed in 0.1.2.

**Comment by trusktr-goin** [comment]
@justinmk You can close this now. :}
